### ISSUE TITLE
Invalid links with no text are generated for lineanchors without linenos

### ISSUE THREAD

**Issue by Kwpolska** [pygments/pygments#2013]
```
$ cat x.py
if 1 == 2:
    print("Hello!")
$ pygmentize -f html -O lineanchors=x x.py
<div class="highlight"><pre><span></span><a id="x-1" name="x-1" href="x-1"></a><span class="k">if</span> <span class="mi">1</span> <span class="o">==</span> <span class="mi">2</span><span class="p">:</span>
<a id="x-2" name="x-2" href="x-2"></a>    <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Hello!&quot;</span><span class="p">)</span>
</pre></div>
```

This change was introduced in #1973. There are two issues with it:

1. The `href` attribute is not valid, since it links to a file and not the anchor. It should refer to `#x-1` for example.
2. I'm not convinced that this behavior is useful. I can access the link with the Tab key, but I can't access it via mouse. This feature would require extra CSS, which the default Pygments CSS does not include. Is the required CSS documented anywhere?

**Comment by Anteru** [comment]
You're right on the missing `#`, but I'm not following the "is that behavior useful" question. There are `id`s assigned so it makes sense to link to them. Is there some use case that is broken now that was working differently previously?

**Comment by Kwpolska** [comment]
It might make sense to link to them — but if you’re linking to them, why not just use line numbers, or something visible? The new feature is not accessible to mouse users and might be confusing for keyboard and screen-reader users, and I don’t think the feature is documented or explained anywhere.

**Comment by ZedThree** [comment]
@Kwpolska See https://github.com/pygments/pygments/pull/1973#issuecomment-1004852001 for some CSS to make the line numbers visible/clickable

I think this is only needed if you're using CSS to display the line numbers, rather than using `linenos` + `anchorlinenos`.
